### PR TITLE
Add compilation services when adding Razor.

### PR DESCRIPTION
**Pull request by NTaylorMullen**
#3383

/cc @pranavkm

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Razor/DependencyInjection/MvcRazorMvcCoreBuilderExtensions.cs [hunk 1]
```diff
@@ -117,6 +118,12 @@ public static IMvcCoreBuilder AddRazorViewEngine(this IMvcCoreBuilder builder)
         // Internal for testing.
         internal static void AddRazorViewEngineServices(IServiceCollection services)
         {
+            if (CompilationServices.Default != null)
+            {
+                services.TryAdd(ServiceDescriptor.Instance(CompilationServices.Default.LibraryExporter));
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Razor/DependencyInjection/MvcRazorMvcCoreBuilderExtensions.cs]: Extra blank line YO

**Comment by rynowak** [comment]
:shipit:

File: src/Microsoft.AspNet.Mvc.Razor/DependencyInjection/MvcRazorMvcCoreBuilderExtensions.cs (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Razor/DependencyInjection/MvcRazorMvcCoreBuilderExtensions.cs]: :8ball:

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**